src/poetry/packages/locker.py

```diff
@@ -11,6 +11,7 @@
 from typing import Iterable
```

Comment:
Changes in this file look good, but I think we should break them out into a separate commit.